server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java

```diff
@@ -113,249 +112,111 @@ private static boolean isValidColumn(ColumnUpdate cu) {
     return validColumnQuals.contains(new ColumnFQ(cu));
   }
 
-  private static ArrayList<Short> addViolation(ArrayList<Short> lst, int violation) {
-    if (lst == null) {
-      lst = new ArrayList<>();
-    }
+  private static void addViolation(ArrayList<Short> lst, int violation) {
     lst.add((short) violation);
-    return lst;
   }
 
-  private static ArrayList<Short> addIfNotPresent(ArrayList<Short> lst, int intViolation) {
-    if (lst == null) {
-      return addViolation(null, intViolation);
-    }
-    short violation = (short) intViolation;
-    if (!lst.contains(violation)) {
-      return addViolation(lst, intViolation);
+  private static void addIfNotPresent(ArrayList<Short> lst, int violation) {
+    if (!lst.contains((short) violation)) {
+      addViolation(lst, violation);
     }
-    return lst;
   }
 
   /*
    * Validates the data file metadata is valid for a StoredTabletFile.
    */
-  private static ArrayList<Short> validateDataFileMetadata(ArrayList<Short> violations,
-      String metadata) {
+  private static void validateDataFileMetadata(ArrayList<Short> violations, String metadata,
+      BulkFileColData bfcValidationData, boolean addToDataFiles, boolean addToLoadedFiles) {
```

Comment:
This could be simplified by returning the StoredTabletFile object.  Then the called can add it to the desired set.


```java
  private static StoredTabletFile validateDataFileMetadata(ArrayList<Short> violations, String metadata) {
         try {
      StoredTabletFile stf = StoredTabletFile.of(metadata);
      return stf;
    } catch (RuntimeException e) {
      addViolation(violations, 9);
    }
 }
```

Reply:
Could do the following instead of returning.  This would avoid having to deal with a null return when an exception is thrown.

```java
  private static void validateDataFileMetadata(ArrayList<Short> violations, String metadata, Consumer<StoredTabletFile> stfConsumer) {
         try {
            stfConsumer.accept(StoredTabletFile.of(metadata));
           } catch (RuntimeException e) {
             addViolation(violations, 9);
          }
 }
```

Reply:
Should have mentioned this earlier.  With the consumer approach could pass method references like :

```java
    validateDataFileMetadata(violations, new String(columnUpdate.getColumnQualifier(), UTF_8),
        bfcValidationData::addDataFile);
```

Reply:
This is nice, added